src/transformers/modeling_flash_attention_utils.py

```diff
@@ -353,61 +244,101 @@ def _prepare_flash_attention_from_position_ids(query, key, value, position_ids):
     # We should use cu_seq_lens instead of position_ids to get the max length since position_ids is not always increasing
     # for some models (e.g. qwen2-vl).
     max_length = cu_seq_lens.diff().max().item()
+    return (query, key, value, indices_q, (cu_seq_lens, cu_seqlens_k), (max_length, max_k))
 
-    return (query, key, value, indices_q, (cu_seq_lens, cu_seq_lens), (max_length, max_length))
 
-
-def prepare_fa2_from_position_ids(*args, **kwargs):
+def _prepare_flash_attention_from_position_ids(query, key, value, position_ids):
     warnings.warn(
-        "The function `prepare_fa2_from_position_ids` in `transformers.modeling_flash_attention_utils` is deprecated and will be removed in a future version. Please use `_prepare_flash_attention_from_position_ids` instead.",
+        "prepare_fa2_from_position_ids is deprecated, use _prepare_flash_attention_from_position_ids",
```

Comment:
I think you mean this one
```suggestion
        "prepare_fa2_from_position_ids is deprecated, use _prepare_from_posids",
```